packages/node_modules/@node-red/runtime/lib/nodes/context/index.js

```diff
@@ -254,6 +272,29 @@ function createContext(id,seed) {
                     if (!storage) {
                         storage = keyParts.store || "_";
                     }
+                    var result = followParentContext(parent, key);
+                    if (result) {
+                        var [ctx, new_key] = result;
+                        if (ctx) {
+                            if (new_key === "") {
```

Comment:
I'm not sure this is the right behaviour - but am happy to discuss and be convinced otherwise.

If a key of `$parent` (or `$parent.$parent` etc) is used, then the key to look up in the targeted context is a blank string. That is not a valid context key so *I think* it should return `undefined`. We shouldn't return the Context object itself.

Reply:
My intension on this behavior is to make `$parent...` scope access faster when frequent access to values in parent context is needed.
However, since `$parent` access seems to be infrequent, I think that it is OK  to make `$parent` access with empty key returns `undefined`.